core/services/ocr2/plugins/llo/integration_test.go

```diff
@@ -0,0 +1,370 @@
+			report, exists := v["report"]
+			if !exists {
+				t.Fatalf("FAIL: expected payload %#v to contain 'report'", v)
+			}
+
+			t.Logf("Got report from oracle %x with format: %d", req.pk, req.req.ReportFormat)
+
+			var r datastreamsllo.Report
+
+			switch req.req.ReportFormat {
+			case uint32(llotypes.ReportFormatJSON):
+				t.Logf("Got report (JSON) from oracle %x: %s", req.pk, string(report.([]byte)))
+				var err error
+				r, err = (datastreamsllo.JSONReportCodec{}).Decode(report.([]byte))
+				require.NoError(t, err, "expected valid JSON")
+			case uint32(llotypes.ReportFormatEVM):
+				t.Logf("Got report (EVM) from oracle %x: 0x%x", req.pk, report.([]byte))
+				var err error
+				r, err = (lloevm.ReportCodec{}).Decode(report.([]byte))
+				require.NoError(t, err, "expected valid EVM encoding")
+			default:
+				t.Fatalf("FAIL: unexpected report format: %q", req.req.ReportFormat)
+			}
+
+			assert.Equal(t, configDigest, r.ConfigDigest)
+			assert.Equal(t, uint64(0x2ee634951ef71b46), r.ChainSelector)
+			assert.GreaterOrEqual(t, r.SeqNr, uint64(1))
+			assert.GreaterOrEqual(t, r.ValidAfterSeconds, testStartTimeStamp)
+			assert.Equal(t, r.ValidAfterSeconds+1, r.ValidUntilSeconds)
+
+			// values
+			defn, exists := channelDefinitions[r.ChannelID]
+			require.True(t, exists, "expected channel ID to be in channelDefinitions")
+
+			require.Equal(t, len(defn.StreamIDs), len(r.Values))
+
+			for i, strmID := range defn.StreamIDs {
+				strm, exists := streamMap[strmID]
+				require.True(t, exists, "invariant violation: expected stream ID to be present")
+				assert.InDelta(t, strm.baseBenchmarkPrice.Int64(), r.Values[i].Int64(), 5000000)
+			}
+
+			assert.False(t, r.Specimen)
+
+			seen[req.TransmitterID()][r.ChannelID] = struct{}{}
+			t.Logf("Got report from oracle %s with channel: %x)", req.TransmitterID(), r.ChannelID)
+
+			if _, exists := seen[req.TransmitterID()]; exists && len(seen[req.TransmitterID()]) == len(channelDefinitions) {
+				t.Logf("All channels reported for oracle with transmitterID %s", req.TransmitterID())
+				delete(seen, req.TransmitterID())
+			}
+			if len(seen) == 0 {
+				break // saw all oracles; success!
+			}
+
+			// bit of a hack here but shouldn't hurt anything, we wanna dump
+			// `seen` before the test ends to aid in debugging test failures
+			if d, ok := t.Deadline(); ok {
+				select {
+				case <-time.After(time.Until(d.Add(-100 * time.Millisecond))):
+					if len(seen) > 0 {
+						t.Fatalf("FAILED: ERROR: missing expected reports: %#v\n", seen)
+					}
+				default:
+				}
+			}
+		}
+	})
+
+	// TODO: test verification
+}
+
+func setConfig(t *testing.T, steve *bind.TransactOpts, backend *backends.SimulatedBackend, verifierContract *channel_verifier.ChannelVerifier, verifierAddress common.Address, nodes []Node, oracles []confighelper.OracleIdentityExtra) ocr2types.ConfigDigest {
+	// Setup config on contract
+	rawOnchainConfig := datastreamsllo.OnchainConfig{}
+	// TODO: Move away from JSON
```

Comment:
Future ticket?

Reply:
It already has one and listed in another comment elsewhere. Will remove.

https://smartcontract-it.atlassian.net/jira/software/c/projects/MERC/issues/MERC-3661